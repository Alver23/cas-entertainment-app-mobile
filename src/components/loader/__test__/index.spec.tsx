// Dependencies
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ActivityIndicator, StatusBar } from 'react-native';

// Components
import { LoaderView } from '../index';

const MemoizeComponent = LoaderView.type;

describe('LoaderView component', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<MemoizeComponent />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render correctly', () => {
    expect(component.find(ActivityIndicator).exists()).toBeTruthy();

    expect(component.find(StatusBar).exists()).toBeTruthy();
  });
});
