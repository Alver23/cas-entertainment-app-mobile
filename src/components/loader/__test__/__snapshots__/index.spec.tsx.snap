// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderView component should save a snapshot of the component 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 3,
      },
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "padding": 10,
      },
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#2089dc"
    hidesWhenStopped={true}
    size="large"
  />
  <StatusBar
    animated={false}
    barStyle="dark-content"
    showHideTransition="fade"
  />
</View>
`;
